### PR TITLE
clean up iptables/ipvs proxy mode descriptions a bit

### PR DESCRIPTION
**Pull request by danwinship**
Noticed while adding the nftables documentation.

filing this against `dev-1.29` because @sftim [said to](https://github.com/kubernetes/website/pull/43588#pullrequestreview-1750988605) but nothing in here is new in 1.29.

### CODE REVIEW

File: content/en/docs/reference/networking/virtual-ips.md [hunk 1]
```diff
@@ -14,6 +14,18 @@ The `kube-proxy` component is responsible for implementing a _virtual IP_
 mechanism for {{< glossary_tooltip term_id="service" text="Services">}}
 of `type` other than
 [`ExternalName`](/docs/concepts/services-networking/service/#externalname).
+Kube-proxy watches the Kubernetes {{< glossary_tooltip
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:17]:
```suggestion
Each instance of kube-proxy watches the Kubernetes {{< glossary_tooltip
```

File: content/en/docs/reference/networking/virtual-ips.md [hunk 2]
```diff
@@ -14,6 +14,18 @@ The `kube-proxy` component is responsible for implementing a _virtual IP_
...
+term_id="control-plane" text="control plane" >}} for the addition and
+removal of Service and EndpointSlice {{< glossary_tooltip
+term_id="object" text="objects." >}} For each Service, kube-proxy
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:20]:
```suggestion
term_id="object" text="objects" >}}. For each Service, kube-proxy
```

File: content/en/docs/reference/networking/virtual-ips.md [hunk 3]
```diff
@@ -14,6 +14,18 @@ The `kube-proxy` component is responsible for implementing a _virtual IP_
...
+calls appropriate APIs (depending on the kube-proxy mode) to configure
+the node to capture traffic to the Service's `clusterIP` and `port`,
+and redirect that traffic to one of the Service's endpoint IPs
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:23]:
```suggestion
and redirect that traffic to one of the Service's endpoints
```
It's a tuple of (address, port, &hellip;) I think, not just an IP address.

File: content/en/docs/reference/networking/virtual-ips.md [hunk 4]
```diff
@@ -14,6 +14,18 @@ The `kube-proxy` component is responsible for implementing a _virtual IP_
...
+(usually a Pod, but possibly an arbitrary user-provided IP). A control
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:24]:
```suggestion
(usually a Pod, but possibly an arbitrary user-provided IP address). A control
```

File: content/en/docs/reference/networking/virtual-ips.md [hunk 5]
```diff
@@ -125,8 +115,10 @@ through a load-balancer, though in those cases the client IP address does get al
 
 #### Optimizing iptables mode performance
 
-In large clusters (with tens of thousands of Pods and Services), the
-iptables mode of kube-proxy may take a long time to update the rules
+In iptables mode, kube-proxy creates a few iptables rules for every
+Service, and a few iptables rules for every endpoint IP. In
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:119]:
```suggestion
Service, and a few iptables rules for each endpoint IP address. In
```

File: content/en/docs/reference/networking/virtual-ips.md [hunk 6]
```diff
@@ -266,7 +255,7 @@ the node before starting kube-proxy.
 
 When kube-proxy starts in IPVS proxy mode, it verifies whether IPVS
 kernel modules are available. If the IPVS kernel modules are not detected, then kube-proxy
-falls back to running in iptables proxy mode.
+will exit with an error.
```
Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:258]:
```suggestion
exits with an error. Kubernetes does offer you some features that you can use to help
with that; for example, you can define an
{{< glossary_tooltip text="init containers" term_id="init-container" >}} that ensures the relevant
module are loaded before starting the container for `kube-proxy`.
```
?

**Review by sftim — COMMENTED** [review]
My thoughts - hope these make sense?

**Comment by katcosgrove** [comment]
Hey @danwinship! If we can land this before I freeze the website on Monday, it's good to go in with dev-1.29.

File: content/en/docs/reference/networking/virtual-ips.md (hunk 2 above)
Comment (danwinship) [reply content/en/docs/reference/networking/virtual-ips.md:20]: oh, good, I had just moved this from somewhere else and was like "wow... that's... a style choice"

File: content/en/docs/reference/networking/virtual-ips.md (hunk 6 above)
Comment (sftim) [reply content/en/docs/reference/networking/virtual-ips.md:258]: Can we document the minimum Linux kernel version for IPVS mode (maybe for different flavors, if that matters)?

I'd assumed it was 2.2.0 or something venerable, that we could assume everyone has.

Comment (sftim) [inline content/en/docs/reference/networking/virtual-ips.md:258]: Try this, then:
```
exits with an error.
```

**Review by sftim — COMMENTED** [review]
/lgtm

A review from SIG Network would be welcome.

**Comment by Shubham82** [comment]
@danwinship The test cases failed due to Netlify build timeout failures.
please re-push the commit.

```
git reset
git commit --amend --no-edit
git push --force-with-lease
```

**Comment by sftim** [comment]
No need to squash for 3 commits.
/remove-label tide/merge-method-squash

**Pull request merged**